configure

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+
+AUTO_QMAIL=/var/qmail
+
+for arg in $*; do
+	case "${arg}" in
+	--prefix=*)
+		AUTO_QMAIL=$(echo "${arg}" | sed 's#^--prefix=##')
+		;;
+	--help)
+		echo "--prefix    install prefix [/var/qmail]"
+		exit 0
+		;;
+	*)
+		echo "unknown argument: ${arg}" >&2
+		exit 1
+		;;
+	esac
+done
+
+[ -z "${CC}" ] && CC=cc
+[ -z "${CFLAGS}" ] && CFLAGS=-O2
+[ -z "${LD}" ] && LD="cc -s"
+[ -z "${AR}" ] && AR=ar
+[ -z "${RANLIB}" ] && RANLIB=ranlib
+
+cat > conf-cc << EOF
+${CC} ${CFLAGS}
+
+This will be used to compile .c files.
+EOF
+
+cat > conf-ld << EOF
+${LD}
+
+This will be used to link .o files into an executable.
+EOF
+
+cat > conf-qmail << EOF
+${AUTO_QMAIL}
+
+This is the qmail home directory. It must be a local directory, not
+shared among machines. This is where qmail queues all mail messages.
+
+The queue (except for bounce message contents) is crashproof, if the
+filesystem guarantees that single-byte writes are atomic and that
+directory operations are synchronous. These guarantees are provided by
+fixed-block filesystems such as UFS and by journaling filesystems. Under
+Linux, make sure that all mail-handling filesystems are mounted with
+synchronous metadata.
+EOF
+
+cat > make-makelib.sh << EOF
+echo 'main="\$1"; shift'
+echo 'rm -f "\$main"'
+echo '${AR} cr "\$main" \${1+"\$@"}'
+echo '${RANLIB} "\$main"'
```

Comment:
Now, we have a script (`./configure`) that generates a script (`make-makelib.sh`) that is used to generate the script (`make-makelib`) that generates the script (`makelib`) that generates the library.

I will propose an extra branch for changing the make-makelib.sh to make AR and RANLIB configurable like CC and LD.